spring-amqp/src/main/java/org/springframework/amqp/support/ConditionalExceptionLogger.java

```diff
@@ -37,4 +39,16 @@ public interface ConditionalExceptionLogger {
 	 */
 	void log(Log logger, String message, Throwable t);
 
+	/**
+	 * Log a consumer restart; debug by default.
+	 * @param logger the logger.
+	 * @param message the message.
+	 * @since 3.1
+	 */
+	default void logRestart(Log logger, Supplier<String> message) {
```

Comment:
Maybe we can use `org.springframework.core.log.LogMessage` instead?
And then we won't need the next `isDebugEnabled()`